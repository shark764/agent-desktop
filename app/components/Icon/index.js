/**
*
* Icon
*
*/

import React, { PropTypes } from 'react';
import config from 'assets/icons/config.png';
import message from 'assets/icons/message.png';
import messageNew from 'assets/icons/message_new.png';
import email from 'assets/icons/email.png';
import emailNew from 'assets/icons/email_new.png';
import connected from 'assets/icons/connected.png';
import notConnected from 'assets/icons/not_connected.png';
import dialpad from 'assets/icons/dialpad.png';
import dialpadHover from 'assets/icons/dialpad_hover.png';

import Radium from 'radium';

function Icon(props) {
  let icon;
  let iconHover;
  let borderRadius;
  switch (props.name) {
    case 'config':
      icon = config;
      break;
    case 'message':
      icon = message;
      break;
    case 'message_new':
      icon = messageNew;
      break;
    case 'email':
      icon = email;
      break;
    case 'email_new':
      icon = emailNew;
      break;
    case 'connected':
      icon = connected;
      break;
    case 'not_connected':
      icon = notConnected;
      break;
    case 'dialpad':
      icon = dialpad;
      iconHover = dialpadHover;
      borderRadius = '50%';
      break;
    default:
      break;
  }

  const styles = {
    base: {
      ':hover': {
        content: iconHover ? `url(${iconHover})` : '',
      },
      borderRadius,
    },
  };

  return (
    <img id={props.id} src={icon} style={[props.style, styles.base]} alt={props.name} />
  );
}

Icon.propTypes = {
  name: PropTypes.string.isRequired,
  style: PropTypes.object,
  id: PropTypes.string.isRequired,
};

export default Radium(Icon);
