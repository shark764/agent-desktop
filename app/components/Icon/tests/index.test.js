/*
 * Copyright Â© 2015-2017 Serenova, LLC. All rights reserved.
 */

import React from 'react';
import { shallow } from 'enzyme';

import Icon, { availableIcons, availableIconsWithActive } from '../index';

describe('<Icon />', () => {
  availableIcons.forEach((name) => {
    describe(`with name ${name}`, () => {
      it('should render correctly', () => {
        const rendered = shallow(<Icon id="mockId" name={name} />);
        expect(rendered).toMatchSnapshot();
      });
    });
  });

  availableIconsWithActive.forEach((name) => {
    describe(`with name ${name} and active prop`, () => {
      it('should render correctly', () => {
        const rendered = shallow(<Icon id="mockId" name={name} active />);
        expect(rendered).toMatchSnapshot();
      });
    });
  });
});
