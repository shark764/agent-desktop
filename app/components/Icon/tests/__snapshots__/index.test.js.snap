// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> with name add_interaction should render correctly 1`] = `
<img
  alt="add_interaction"
  data-radium={true}
  id="mockId"
  src="test-file-stub"
  style={
    Object {
      "cursor": "pointer",
    }
  }
/>
`;

exports[`<Icon /> with name arrow_return should render correctly 1`] = `
<img
  alt="arrow_return"
  data-radium={true}
  id="mockId"
  src="test-file-stub"
  style={
    Object {
      "cursor": "pointer",
    }
  }
/>
`;

exports[`<Icon /> with name arrow_up_down should render correctly 1`] = `
<img
  alt="arrow_up_down"
  data-radium={true}
  id="mockId"
  src="test-file-stub"
  style={
    Object {
      "cursor": "pointer",
    }
  }
/>
`;

exports[`<Icon /> with name attachment should render correctly 1`] = `
<img
  alt="attachment"
  data-radium={true}
  id="mockId"
  src="test-file-stub"
  style={
    Object {
      "cursor": "pointer",
      "height": "16px",
      "width": "16px",
    }
  }
/>
`;

exports[`<Icon /> with name caret should render correctly 1`] = `
<img
  alt="caret"
  data-radium={true}
  id="mockId"
  src="test-file-stub"
  style={
    Object {
      "cursor": "pointer",
      "height": "6px",
      "width": "10px",
    }
  }
/>
`;

exports[`<Icon /> with name caret_white should render correctly 1`] = `
<img
  alt="caret_white"
  data-radium={true}
  id="mockId"
  src="test-file-stub"
  style={
    Object {
      "cursor": "pointer",
      "height": "10px",
      "width": "18px",
    }
  }
/>
`;

exports[`<Icon /> with name checkMark should render correctly 1`] = `
<img
  alt="checkMark"
  data-radium={true}
  id="mockId"
  src="test-file-stub"
  style={
    Object {
      "cursor": "pointer",
      "height": "25px",
    }
  }
/>
`;

exports[`<Icon /> with name checkStatus should render correctly 1`] = `
<img
  alt="checkStatus"
  data-radium={true}
  id="mockId"
  src="test-file-stub"
  style={
    Object {
      "cursor": "pointer",
      "height": "17px",
    }
  }
/>
`;

exports[`<Icon /> with name close should render correctly 1`] = `
<img
  alt="close"
  data-radium={true}
  id="mockId"
  src="test-file-stub"
  style={
    Object {
      "cursor": "pointer",
    }
  }
/>
`;

exports[`<Icon /> with name config should render correctly 1`] = `
<img
  alt="config"
  data-radium={true}
  id="mockId"
  src="test-file-stub"
  style={
    Object {
      "cursor": "pointer",
    }
  }
/>
`;

exports[`<Icon /> with name connected should render correctly 1`] = `
<img
  alt="connected"
  data-radium={true}
  id="mockId"
  src="test-file-stub"
  style={
    Object {
      "cursor": "pointer",
    }
  }
/>
`;

exports[`<Icon /> with name dialpad and active prop should render correctly 1`] = `
<span
  className="active"
  data-radium={true}
  id="mockId"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "backgroundImage": "url(test-file-stub)",
      "backgroundPosition": "-80px 0",
      "borderRadius": "50%",
      "cursor": "pointer",
      "display": "block",
      "height": "40px",
      "width": "40px",
    }
  }
/>
`;

exports[`<Icon /> with name dialpad should render correctly 1`] = `
<span
  className=""
  data-radium={true}
  id="mockId"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "backgroundImage": "url(test-file-stub)",
      "borderRadius": "50%",
      "cursor": "pointer",
      "display": "block",
      "height": "40px",
      "width": "40px",
    }
  }
/>
`;

exports[`<Icon /> with name dialpad_dark and active prop should render correctly 1`] = `
<span
  className="active"
  data-radium={true}
  id="mockId"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "backgroundImage": "url(test-file-stub)",
      "backgroundPosition": "-80px 0",
      "backgroundSize": "120px 40px",
      "borderRadius": "50%",
      "cursor": "pointer",
      "display": "block",
      "height": "40px",
      "width": "40px",
    }
  }
/>
`;

exports[`<Icon /> with name dialpad_dark should render correctly 1`] = `
<span
  className=""
  data-radium={true}
  id="mockId"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "backgroundImage": "url(test-file-stub)",
      "backgroundSize": "120px 40px",
      "borderRadius": "50%",
      "cursor": "pointer",
      "display": "block",
      "height": "40px",
      "width": "40px",
    }
  }
/>
`;

exports[`<Icon /> with name email should render correctly 1`] = `
<img
  alt="email"
  data-radium={true}
  id="mockId"
  src="test-file-stub"
  style={
    Object {
      "cursor": "pointer",
      "height": "15px",
      "width": "18px",
    }
  }
/>
`;

exports[`<Icon /> with name email_dark should render correctly 1`] = `
<img
  alt="email_dark"
  data-radium={true}
  id="mockId"
  src="test-file-stub"
  style={
    Object {
      "cursor": "pointer",
      "height": "17px",
      "width": "20px",
    }
  }
/>
`;

exports[`<Icon /> with name email_new should render correctly 1`] = `
<img
  alt="email_new"
  data-radium={true}
  id="mockId"
  src="test-file-stub"
  style={
    Object {
      "cursor": "pointer",
      "height": "15px",
      "width": "18px",
    }
  }
/>
`;

exports[`<Icon /> with name end_call_resource should render correctly 1`] = `
<span
  className=""
  data-radium={true}
  id="mockId"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "backgroundImage": "url(test-file-stub)",
      "backgroundSize": "72px 24px",
      "borderRadius": "50%",
      "cursor": "pointer",
      "display": "block",
      "height": "24px",
      "width": "24px",
    }
  }
/>
`;

exports[`<Icon /> with name endCall should render correctly 1`] = `
<span
  className=""
  data-radium={true}
  id="mockId"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "backgroundImage": "url(test-file-stub)",
      "borderRadius": "50%",
      "cursor": "pointer",
      "display": "block",
      "height": "40px",
      "width": "40px",
    }
  }
/>
`;

exports[`<Icon /> with name hold and active prop should render correctly 1`] = `
<span
  className="active"
  data-radium={true}
  id="mockId"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "backgroundImage": "url(test-file-stub)",
      "backgroundPosition": "-80px 0",
      "borderRadius": "50%",
      "cursor": "pointer",
      "display": "block",
      "height": "40px",
      "width": "40px",
    }
  }
/>
`;

exports[`<Icon /> with name hold should render correctly 1`] = `
<span
  className=""
  data-radium={true}
  id="mockId"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "backgroundImage": "url(test-file-stub)",
      "borderRadius": "50%",
      "cursor": "pointer",
      "display": "block",
      "height": "40px",
      "width": "40px",
    }
  }
/>
`;

exports[`<Icon /> with name hold_resource and active prop should render correctly 1`] = `
<span
  className="active"
  data-radium={true}
  id="mockId"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "backgroundImage": "url(test-file-stub)",
      "backgroundPosition": "-48px 0",
      "backgroundSize": "72px 24px",
      "borderRadius": "50%",
      "cursor": "pointer",
      "display": "block",
      "height": "24px",
      "width": "24px",
    }
  }
/>
`;

exports[`<Icon /> with name hold_resource should render correctly 1`] = `
<span
  className=""
  data-radium={true}
  id="mockId"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "backgroundImage": "url(test-file-stub)",
      "backgroundSize": "72px 24px",
      "borderRadius": "50%",
      "cursor": "pointer",
      "display": "block",
      "height": "24px",
      "width": "24px",
    }
  }
/>
`;

exports[`<Icon /> with name message should render correctly 1`] = `
<img
  alt="message"
  data-radium={true}
  id="mockId"
  src="test-file-stub"
  style={
    Object {
      "cursor": "pointer",
      "height": "15px",
      "width": "18px",
    }
  }
/>
`;

exports[`<Icon /> with name message_dark should render correctly 1`] = `
<img
  alt="message_dark"
  data-radium={true}
  id="mockId"
  src="test-file-stub"
  style={
    Object {
      "cursor": "pointer",
      "height": "20px",
      "width": "20px",
    }
  }
/>
`;

exports[`<Icon /> with name message_new should render correctly 1`] = `
<img
  alt="message_new"
  data-radium={true}
  id="mockId"
  src="test-file-stub"
  style={
    Object {
      "cursor": "pointer",
      "height": "15px",
      "width": "18px",
    }
  }
/>
`;

exports[`<Icon /> with name mute and active prop should render correctly 1`] = `
<span
  className="active"
  data-radium={true}
  id="mockId"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "backgroundImage": "url(test-file-stub)",
      "backgroundPosition": "-80px 0",
      "borderRadius": "50%",
      "cursor": "pointer",
      "display": "block",
      "height": "40px",
      "width": "40px",
    }
  }
/>
`;

exports[`<Icon /> with name mute should render correctly 1`] = `
<span
  className=""
  data-radium={true}
  id="mockId"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "backgroundImage": "url(test-file-stub)",
      "borderRadius": "50%",
      "cursor": "pointer",
      "display": "block",
      "height": "40px",
      "width": "40px",
    }
  }
/>
`;

exports[`<Icon /> with name not_connected should render correctly 1`] = `
<img
  alt="not_connected"
  data-radium={true}
  id="mockId"
  src="test-file-stub"
  style={
    Object {
      "cursor": "pointer",
    }
  }
/>
`;

exports[`<Icon /> with name resources and active prop should render correctly 1`] = `
<span
  className="active"
  data-radium={true}
  id="mockId"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "backgroundImage": "url(test-file-stub)",
      "backgroundPosition": "-80px 0",
      "backgroundSize": "120px 40px",
      "borderRadius": "50%",
      "cursor": "pointer",
      "display": "block",
      "height": "40px",
      "width": "40px",
    }
  }
/>
`;

exports[`<Icon /> with name resources should render correctly 1`] = `
<span
  className=""
  data-radium={true}
  id="mockId"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "backgroundImage": "url(test-file-stub)",
      "backgroundSize": "120px 40px",
      "borderRadius": "50%",
      "cursor": "pointer",
      "display": "block",
      "height": "40px",
      "width": "40px",
    }
  }
/>
`;

exports[`<Icon /> with name script should render correctly 1`] = `
<img
  alt="script"
  data-radium={true}
  id="mockId"
  src="test-file-stub"
  style={
    Object {
      "cursor": "pointer",
      "height": "22px",
      "width": "26px",
    }
  }
/>
`;

exports[`<Icon /> with name search should render correctly 1`] = `
<img
  alt="search"
  data-radium={true}
  id="mockId"
  src="test-file-stub"
  style={
    Object {
      "cursor": "pointer",
      "height": "16px",
      "width": "16px",
    }
  }
/>
`;

exports[`<Icon /> with name transfer and active prop should render correctly 1`] = `
<span
  className="active"
  data-radium={true}
  id="mockId"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "backgroundImage": "url(test-file-stub)",
      "backgroundPosition": "-80px 0",
      "borderRadius": "50%",
      "cursor": "pointer",
      "display": "block",
      "height": "40px",
      "width": "40px",
    }
  }
/>
`;

exports[`<Icon /> with name transfer should render correctly 1`] = `
<span
  className=""
  data-radium={true}
  id="mockId"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "backgroundImage": "url(test-file-stub)",
      "borderRadius": "50%",
      "cursor": "pointer",
      "display": "block",
      "height": "40px",
      "width": "40px",
    }
  }
/>
`;

exports[`<Icon /> with name transfer_dark and active prop should render correctly 1`] = `
<span
  className="active"
  data-radium={true}
  id="mockId"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "backgroundImage": "url(test-file-stub)",
      "backgroundPosition": "-80px 0",
      "backgroundSize": "120px 40px",
      "borderRadius": "50%",
      "cursor": "pointer",
      "display": "block",
      "height": "40px",
      "width": "40px",
    }
  }
/>
`;

exports[`<Icon /> with name transfer_dark should render correctly 1`] = `
<span
  className=""
  data-radium={true}
  id="mockId"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "backgroundImage": "url(test-file-stub)",
      "backgroundSize": "120px 40px",
      "borderRadius": "50%",
      "cursor": "pointer",
      "display": "block",
      "height": "40px",
      "width": "40px",
    }
  }
/>
`;

exports[`<Icon /> with name transfer_resource should render correctly 1`] = `
<span
  className=""
  data-radium={true}
  id="mockId"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "backgroundImage": "url(test-file-stub)",
      "backgroundSize": "72px 24px",
      "borderRadius": "50%",
      "cursor": "pointer",
      "display": "block",
      "height": "24px",
      "width": "24px",
    }
  }
/>
`;

exports[`<Icon /> with name voice should render correctly 1`] = `
<img
  alt="voice"
  data-radium={true}
  id="mockId"
  src="test-file-stub"
  style={
    Object {
      "cursor": "pointer",
      "height": "20px",
      "width": "16px",
    }
  }
/>
`;

exports[`<Icon /> with name voice_dark should render correctly 1`] = `
<img
  alt="voice_dark"
  data-radium={true}
  id="mockId"
  src="test-file-stub"
  style={
    Object {
      "cursor": "pointer",
      "height": "20px",
      "width": "16px",
    }
  }
/>
`;

exports[`<Icon /> with name voice_white should render correctly 1`] = `
<img
  alt="voice_white"
  data-radium={true}
  id="mockId"
  src="test-file-stub"
  style={
    Object {
      "cursor": "pointer",
      "height": "20px",
      "width": "16px",
    }
  }
/>
`;
